src/vm.rs

```diff
@@ -82,26 +82,30 @@ impl EraVM {
     }
 
     /// Run a vm program with a given bytecode.
-    pub fn run_program_with_custom_bytecode(&mut self) -> (ExecutionOutput, BlobSaverTracer) {
-        self.run_opcodes()
+    pub fn run_program_with_custom_bytecode(
+        &mut self,
+        tracer: Option<&mut dyn Tracer>,
```

Comment:
I agree with keeping the existing function intact to avoid breaking changes. I've added new functions, `run_program_with_test_encode_and_tracer` & `run_program_with_custom_bytecode_and_tracer`, names might be long but I don't think there is an alternative.

Let me know if this works!